src/pk/asn1/der/utf8/der_length_utf8_string.c

```diff
@@ -27,11 +27,17 @@ unsigned long der_utf8_charsize(const wchar_t c)
       return 1;
    } else if (c <= 0x7FF) {
       return 2;
+#if __WCHAR_MAX__ == 0xFFFF
+   } else {
+      return 3;
```

Comment:
IMO for code point `0xFFFF` you will need 3 bytes in `UTF-8` encoding:

```
code points = 00000000..0000007F, max_bitlen =  7, representation = byte1(0xxxxxxx)
code points = 00000080..000007FF, max_bitlen = 11, representation = byte1(110xxxxx) byte2(10xxxxxx)
code points = 00000800..0000FFFF, max_bitlen = 16, representation = byte1(1110xxxx) byte2(10xxxxxx) byte3(10xxxxxx)
code points = 00010000..0010FFFF, max_bitlen = 21, representation = byte1(11110xxx) byte2(10xxxxxx) byte3(10xxxxxx) byte4(10xxxxxx)
```

https://en.wikipedia.org/wiki/UTF-8